src/main.ts

```diff
@@ -116,7 +116,9 @@ export function generateFile(ctx: Context, fileDesc: FileDescriptorProto): [stri
         }
         if (options.outputJsonMethods) {
           staticMethods.push(generateFromJson(ctx, fullName, message));
-          staticMethods.push(generateFromPartial(ctx, fullName, message));
+          if (options.outputPartialMethods) {
```

Comment:
Per other comment in the readme, let's move this check to be a sibling of the `if outputJsonMethods` block.